processing/src/main/java/io/druid/query/filter/BoundDimFilter.java

```diff
@@ -385,19 +385,20 @@ private void initLongData()
           if (hasLowerBound()) {
             final Long lowerLong = GuavaUtils.tryParseLong(lower);
             if (lowerLong == null) {
-              matchesAnything = false;
-              return;
+              // Unparseable values fall before all actual numbers, so all numbers will match the lower bound.
+              hasLowerLongBoundVolatile = false;
+            } else {
+              hasLowerLongBoundVolatile = true;
+              lowerLongBoundVolatile = lowerLong;
             }
-
-            hasLowerLongBoundVolatile = true;
-            lowerLongBoundVolatile = lowerLong;
           } else {
             hasLowerLongBoundVolatile = false;
```

Comment:
@himanshug I think making things "extra" safe by adding unnecessary volatiles adds confusion because when people see this code, they try to understand "why volatile is present here? It seems unnecessary. Am I missing something?" And waste time double-checking themselves